src/main/java/com/example/bookshop/controller/BookController.java

```diff
@@ -19,46 +22,55 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Book management", description = "Endpoints for managing books")
 @RequiredArgsConstructor
 @RestController
 @RequestMapping("/books")
 public class BookController {
     private final BookService bookService;
 
+    @Operation(summary = "find all books where authors with certain name",
+                description = "find all books by athours")
     @GetMapping("/author")

```

Comment:
You know, I'm really not sure about this enpoint. The endpoint should represent a model resource (for example, if the book includes some other model, then this internal model will be one of the endpoints. That is, a kind of access chain) Author is just one of the non-unique fields. You have an endpoint to search for, and author can be one of the search parameters (I'll leave it to the mentors' experience)

Reply:
you can remove the "author" mapping here and pass just parameter